File: solutions/javascript/AddTwoNumbers/AddTwoNumbers.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
    var a = l1.val + l2.val;
    var b = parseInt(a / 10);
    var res = new ListNode(a % 10);
    var xxx = res;
    var n1 = l1.next;
    var n2 = l2.next;
    while (true) {
        a = n1 === null ? b : n1.val + b;
        a = n2 === null ? a : n2.val + a;
        if (a === 0 && n1 === null && n2 === null) {
            break;
        } else if (n1 === null && n2 === null) {
            xxx.next = new ListNode(a);
            break;
        } else {
            xxx.next = new ListNode(a % 10);
            xxx = xxx.next;
            b = parseInt(a / 10);
            n1 = n1 === null ? null : n1.next;
            n2 = n2 === null ? null : n2.next;
        }
    }
    return res;
};

File: solutions/java/AddTwoNumbers/AddTwoNumbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        int a = l1.val + l2.val;
        int b = a/10;
        ListNode head = new ListNode(a % 10);
        ListNode current = head;
        ListNode n1 = l1.next;
        ListNode n2 = l2.next;
        while (null != n1 || null != n2) {
            a = null == n1 ? b : n1.val + b;
            a = null == n2 ? a : n2.val + a;
            current.next = new ListNode(a % 10);
            current = current.next;
            b = a / 10;
            n1 = null == n1 ? null : n1.next;
            n2 = null == n2 ? null : n2.next;
        }
        if (b > 0) {
            current.next = new ListNode(1);
        }
        return head;
    }
}